src/plugins/anki/anki-plugin.tsx

```diff
@@ -0,0 +1,291 @@
+import { Button } from "@components/Button";
+import {
+  AnoriPlugin,
+  WidgetConfigurationScreenProps,
+  OnCommandInputCallback,
+  WidgetRenderProps,
+  WidgetDescriptor,
+} from "@utils/user-data/types";
+import "./styles.scss";
+import { getAllWidgetsByPlugin } from "@utils/plugin";
+import { translate } from "@translations/index";
+import { useTranslation } from "react-i18next";
+import { useEffect, useState } from "react";
+import { Select } from "@components/Select";
+
+type AnkiPluginWidgetConfigType = {
+  deckName: string;
+  deckId: number;
+};
+
+function invoke(action: string, version: number, params = {}) {
+  return new Promise((resolve, reject) => {
+    const xhr = new XMLHttpRequest();
+    xhr.addEventListener("error", () => reject("failed to issue request"));
+    xhr.addEventListener("load", () => {
+      try {
+        const response = JSON.parse(xhr.responseText);
+        if (Object.getOwnPropertyNames(response).length != 2) {
+          throw "response has an unexpected number of fields";
+        }
+        if (!Object.prototype.hasOwnProperty.call(response, "error")) {
+          throw "response is missing required error field";
+        }
+        if (!Object.prototype.hasOwnProperty.call(response, "result")) {
+          throw "response is missing required result field";
+        }
+        if (response.error) {
+          throw response.error;
+        }
+        resolve(response.result);
+      } catch (e) {
+        reject(e);
+      }
+    });
+
+    xhr.open("POST", "http://127.0.0.1:8765");
+    xhr.send(JSON.stringify({ action, version, params }));
+  });
+}
+
+const WidgetConfigScreen = ({
+  saveConfiguration,
+  currentConfig,
+}: WidgetConfigurationScreenProps<AnkiPluginWidgetConfigType>) => {
+  const [deckName, setDeckName] = useState(
+    currentConfig?.deckName ?? "Default",
+  );
+  const [deckId, setDeckId] = useState(currentConfig?.deckId ?? 1);
+  const [decks, setDecks] = useState<string[]>([]);
+
+  const onConfirm = () => {
+    saveConfiguration({
+      deckName: deckName,
+      deckId: deckId,
+    });
+  };
+
+  const { t } = useTranslation();
+
+  invoke("deckNames", 6).then((data: string[]) => {
```

Comment:
You shouldn't make request directly inside render, paired with `setDecks`, this will make component re-rendering infinitely. It will be better to make request in `useEffect` with empty dependency array